### PR TITLE
fix(gc): harden job-store GC, liveness, and meta writes

### LINKED ISSUE #27: GC and job store robustness
- [ ] GarbageCollect skips jobs whose meta fails to load, silently and forever (internal/manager/manager.go:266-268); RestoreGate does the same (352-358) despite its fail-closed doc; store.Remove errors are discarded without a log line (276-278). Orphan dirs accumulate unboundedly (sources: crash between MkdirAll and the non-atomic initial meta write in Create, internal/jobstore/store.go:81-93, or a partial RemoveAll).
- [ ] GC lacks the sentinel re-read Status performs deliberately (manager.go:273 vs status.go:54-64): a job that writes its sentinel and exits between GC's two checks reads as "not terminal, not alive" and gets removed with unread results. Reachable because job timeouts are uncapped.
- [ ] Store.Dir is the only Store method without the validJobID guard (store.go:191). Safe today because the only caller Loads first; fragile for the next caller. TestRejectsUnsafeJobID covers every method except Dir.
- [ ] SetConversationID's doc claims it cannot clobber a concurrent meta update, but UpdateMeta does not take s.mu (store.go:158-180 vs the mutex doc at store.go:53). Either lock UpdateMeta too or weaken the doc.
- [ ] Liveness fails open when the recorded BootID is empty (internal/manager/liveness_linux.go:35 skips the boot check; manager.go:178 can persist "" when boot_id was unreadable). A recycled PID after reboot can then read as alive, keeping the job uncollectable and the gate key held. Treat empty BootID as fail-closed for the boot check.

Plan: log skipped ids and Remove failures; re-check the sentinel before removal; reap meta-less dirs older than TTL by dir mtime; use the temp+rename pattern in Create; guard Dir; settle the mutex question.

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

Robustness fixes for garbage collection, the job store, and liveness detection (issue #27).

**GarbageCollect**
- Reap meta-less orphan dirs (a crash between `Create`'s `MkdirAll` and the meta write, or a partial `RemoveAll`) once older than the TTL, judged by dir mtime. Only `fs.ErrNotExist` counts as an orphan; a transient or corrupt-meta read error is logged and kept, so a valid long-running job (whose dir mtime is old because writes to out/err do not bump it) is never deleted out from under its supervisor.
- Re-read the exit-code sentinel after `processAlive` returns false, so a job that finished between the two checks is kept this sweep instead of removed with results unread. Also keep a terminal job while its conversation-id capture is still pending (`CapturePending`): the supervisor writes the sentinel before exiting, so removing the dir mid-capture would fail `SetConversationID` and lose the captured id.
- Log skipped unreadable jobs and `store.Remove` failures instead of dropping them silently. `RestoreGate` logs its skips too.

**Job store**
- `Store.Dir` now returns `(string, error)` and guards the id with `validJobID`, so a client-supplied `job_id` cannot escape the store root even for a caller that does not `Load` first.
- `UpdateMeta` takes `s.mu` (via a shared `updateMetaLocked` that `SetConversationID` also calls, avoiding a re-lock deadlock), so `SetConversationID`'s read-modify-write can no longer be clobbered by a concurrent `UpdateMeta`. `Create` now writes `meta.json` with the same temp+rename+fsync helper, so a crash mid-write cannot leave a torn or zero-length `meta.json`.

**Liveness**
- `processAlive` treats a recorded boot id that does not match the current one as not-alive, including an empty recorded id vs a readable current one (boot_id unreadable at start, then a reboot): a recycled PID can no longer read as alive and strand the gate key. When boot_id is unreadable host-wide (e.g. a restricted container) both are empty and compare equal, so liveness still works via the PID and start-time checks.

## Related Issues

Closes #27

## Test Plan

- [x] `go test ./... -race`
- [x] `go vet ./...`, `GOOS=darwin go vet ./...`, `GOOS=windows go build ./...`
- [x] `gofmt -l .` clean, `golangci-lint run ./...` 0 issues
- [x] New tests cover: orphan reap by mtime, corrupt-meta kept, sentinel re-read, capture-pending kept, empty-BootID fail-closed, `Dir` rejects unsafe ids
- [x] Reviewed by an independent Gemini 3.1 Pro pass; its findings (transient-error reaping, container boot_id, capture-window removal) were folded in

## Summary by CodeRabbit

* **Bug Fixes**
  * Job metadata now uses atomic writes to prevent partial-write corruption.
  * Garbage collection more safely handles expired/missing/unreadable job data and defers removal when capture is pending.
  * Store directory lookup now validates job IDs and surfaces errors to callers.
  * Stricter boot-ID checks for cross-reboot liveness and improved logging for unreadable metadata.

* **Tests**
  * Added unit tests covering GC edge cases, liveness boot-ID behavior, and store directory validation.

### CODE REVIEW

File: internal/manager/liveness_linux_test.go [hunk 1]
```diff
@@ -78,6 +78,24 @@ func TestProcessAliveStartTimeCheck(t *testing.T) {
...
+	// An empty recorded boot id cannot be confirmed to belong to the current boot
+	// (boot_id was unreadable when the job started). After a reboot a recycled pid
+	// could otherwise pass the kill and start-time checks and read as alive, keeping
+	// the dead job uncollectable and its gate key held forever. Fail closed: an empty
+	// boot id reads as not alive even for a live, exactly-matching pid.
+	meta := jobstore.Meta{PID: pid, BootID: "", StartTimeTicks: actual}
+	if m.processAlive(meta) {
+		t.Error("empty BootID must fail closed (read as not alive)")
+	}
+}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/manager/liveness_linux_test.go:104]: Good catch, this would have been a non-hermetic failure in restricted containers. Fixed in a49cb7a by skipping the test when readBootID() returns "": with no current boot id to compare against, an empty recorded BootID correctly falls through to the PID and start-time checks, so the fail-closed assertion does not apply there.
